### DISCUSSION TITLE
"Use subject"

### DISCUSSION THREAD

**Discussion by glaucocustodio** [evenbetterspecs/evenbetterspecs.github.io#6] [General]
What are your thoughts on https://evenbetterspecs.github.io/#use-subject?

**Comment by zacharydanger** [comment]
I think this bit

```ruby
describe '#add' do
  it 'sums two numbers' do
    subject = described_class.new(5)

    expect(subject.add(1, 2)).to eq(8)
  end
end
```

should be

```ruby
describe '#add' do
  subject { described_class.new(5) }

  it 'sums two numbers' do
    expect(subject.add(1, 2)).to eq(8)
  end
end
```

to follow [the RSpec docs on explicit subjects](https://rspec.info/features/3-12/rspec-core/subject/explicit-subject/).

**Reply by glaucocustodio** [reply]
the drawback of this approach is that if you need to pass a variable to `described_class.new` you would need to use `let/let!` - which are [not recommended](https://evenbetterspecs.github.io/#lets-not)
